skorch/net.py

```diff
@@ -296,13 +297,19 @@ def on_train_end(self, net,
     # pylint: disable=unused-argument
     def on_epoch_begin(self, net,
                        dataset_train=None, dataset_valid=None, **kwargs):
-        self.history.new_epoch()
-        self.history.record('epoch', len(self.history))
+        epoch_completed = True
+        with suppress(KeyError):
+            epoch_completed = self.history[-1, 'epoch_completed']
```

Comment:
would `epoch_completed = self.history[-1].get('epoch_completed', False)` not suffice?